src/ILCompiler.ReadyToRun/src/ObjectWriter/R2RPEBuilder.cs

```diff
@@ -269,7 +269,22 @@ protected override BlobBuilder SerializeSection(string name, SectionLocation loc
                     else
                     {
                         sectionDataBuilder = new BlobBuilder();
-                        sectionDataBuilder.WriteBytes(inputSectionReader.StartPointer, bytesToRead);
+                        
+                        int offset = _peReader.PEHeaders.PEHeader.CorHeaderTableDirectory.RelativeVirtualAddress - location.RelativeVirtualAddress;
+                        
+                        if (offset >= 0 && offset < bytesToRead)
+                        {
+                            // Patch CorHeader.Flags at offset 16 by removing the ILOnly flag, otherwise the Windows PE loader
+                            // doesn't relocate the image and sets all sections to be non-executable and read-only.
+                            offset += 16;
+                            sectionDataBuilder.WriteBytes(inputSectionReader.CurrentPointer, offset);
+                            inputSectionReader.Offset = offset;
+                            uint corFlags = inputSectionReader.ReadUInt32();
+                            corFlags &= ~(uint)CorFlags.ILOnly;
```

Comment:
We want to also OR in CorFlags.ILLibrary for it to be recognizable by the runtime as being eligible for ready-to-run.